src/main/java/com/fasterxml/jackson/databind/deser/SettableBeanProperty.java

```diff
@@ -637,6 +663,23 @@ protected void _throwAsIOE(JsonParser p, Exception e, Object value) throws IOExc
         _throwAsIOE(p, e);
     }
 
+    /**
+     * @since 2.7
+     */
+    protected IOException _throwAsIOE(JsonParser p, Exception e, boolean wrapException) throws IOException
+    {
+        ClassUtil.throwIfIOE(e);
+        ClassUtil.throwIfRTE(e);
+
+        if (wrapException) {
+            // let's wrap the innermost problem
+            Throwable th = ClassUtil.getRootCause(e);
+            throw JsonMappingException.from(p, ClassUtil.exceptionMessage(th), th);
+        } else {
+            throw JsonMappingException.from(p, ClassUtil.exceptionMessage(e.getCause()), e.getCause());
```

Comment:
Calling `getCause()` because exception is InvocationTarget exception thrown from a setter method.
There maybe other cases? or nah